src/ucp/api/ucp.h

```diff
@@ -1552,6 +1555,32 @@ ucs_status_t ucp_ep_create(ucp_worker_h worker, const ucp_ep_params_t *params,
                            ucp_ep_h *ep_p);
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief Modify endpoint parameters.
+ *
+ * This routine modifies @ref ucp_ep_h "endpoint" created by @ref ucp_ep_create
+ * or @ref ucp_listener_accept_callback_t.
+ *
+ * @param [in]  ep          A handle to the endpoint.
+ * @param [in]  params      User defined @ref ucp_ep_params_t configurations
+ *                          for the @ref ucp_ep_h "UCP endpoint".
+ *
+ * @return UCS_OK           - The endpoint is modified successfully.
+ * @return UCS_PTR_IS_ERR(_ptr) - The reconfiguration failed and an error code
+ *                                indicates the status. However, resources are
```

Comment:
i think we don't want to kill the endpoint, but just keep the original configuration, right?

Reply:
I'm not sure :) this is possible only when reconfiguration was requested for local parameters. Yes, currently, we say that "remote" parameters are not reconfigurable and the only error that can be returned here is `UCS_ERR_INVALID_PARAM` and EP will not be changed. Maybe add this is as an exception? I also thought to add a note that if the function returned "request", the endpoint cannot be used until its completion. @yosefe  What do you think?

Reply:
i think the endpoint can be used but requests will be pending, as with connection establishment
also if there is failure the config remote params the endpoint can be moved to error state

Reply:
makes sense, i'll fix the description like "in case if error is returned, the endpoint is not changed."